src/accelerate/utils/imports.py

```diff
@@ -77,19 +77,28 @@ def is_fp8_available():
     return _is_package_available("transformer_engine")
 
 
+def is_cuda_available():
+    "Checks if `cuda` is available via an `nvml-based` check that won't trigger the drivers"
+    os.environ["PYTORCH_NVML_BASED_CUDA_CHECK"] = str(1)
+    available = torch.cuda.is_available()
+    del os.environ["PYTORCH_NVML_BASED_CUDA_CHECK"]
```

Comment:
Let's use `try`/`finally` to make sure we cleanup the change in the environment in the (unlikely) case we get an error.

Reply:
Not sure why the use of `patch_environment` here was reverted, as it would take care of that. Also, why `str(1)` and not `"1"`?

Reply:
Resulted in a cyclical import issue @BenjaminBossan